z3/snap.py

```diff
@@ -545,6 +553,22 @@ def parse_args():
     return parser.parse_args()
 
 
+class SoftError(Exception):
```

Comment:
Maybe it would be cleaner to move those custom exception in a separate module or to group them. I see that there is another custom exception `IntegrityError` at the beginning of the file.